test/topology/conftest.py

```diff
@@ -168,18 +180,44 @@ async def manager_internal(event_loop, request):
     await manager_int.stop()  # Stop client session and close driver after last test
 
 @pytest.fixture(scope="function")
-async def manager(request, manager_internal):
+async def manager(request, manager_internal, record_property, mode):
     """Per test fixture to notify Manager client object when tests begin so it can
     perform checks for cluster state.
     """
     test_case_name = request.node.name
-    await manager_internal.before_test(test_case_name)
+    run_id = request.config.getoption('run_id')
+    tmp_dir = pathlib.Path(request.config.getoption('tmpdir'))
+    xml_path = pathlib.Path(request.config.getoption('xmlpath'))
+    suite_testpy_log = tmp_dir / mode / f"{xml_path.name[:-10]}.log"
+    test_log = tmp_dir / mode / f"{suite_testpy_log.name[:-4]}.{test_case_name}.log"
```

Comment:
Those `-10` and `-4` magic numbers should have a comment.

Better yet if there was another way to calculate it without relying on hardcoded lengths. What is this doing exactly? Are we trying to cut of some particular suffix? Are we cutting of everything after the last dot, or something like that?

Reply:
Or:

https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem